crates/matrix-sdk-ui/tests/integration/notification.rs

```diff
@@ -0,0 +1,124 @@
+use futures_util::{pin_mut, StreamExt as _};
+use matrix_sdk_test::async_test;
+use matrix_sdk_ui::notifications::NotificationApi;
+
+use crate::{logged_in_client, sync_then_assert_request_and_fake_response, SlidingSyncMatcher};
+
+#[async_test]
+async fn test_smoke_test_notification_api() -> anyhow::Result<()> {
+    let (client, server) = logged_in_client().await;
+
+    let notification_api = NotificationApi::new("notifs".to_owned(), client).await?;
+
+    let notification_stream = notification_api.sync();
+    pin_mut!(notification_stream);
+
+    // Requests enable the e2ee and to_device extensions on the first run.
+    sync_then_assert_request_and_fake_response! {
+        [server, notification_stream]
+        assert request = {
+            "conn_id": "notifs",
+            "extensions": {
+                "e2ee": {
+                    "enabled": true
+                },
+                "to_device": {
+                    "enabled": true
+                }
+            }
+        },
+        respond with = {
+            "pos": "0"
+        },
+    };
+
+    // The request then passes the `pos`ition marker to the next request, as usual
+    // in sliding sync. The extensions haven't changed, so they're not updated
+    // (sticky parameters ftw).
+    sync_then_assert_request_and_fake_response! {
+        [server, notification_stream]
+        assert request = {
+            "conn_id": "notifs",
+        },
+        respond with = {
+            "pos": "1",
+            "extensions": {
+                "to_device": {
+                    "next_batch": "nb0"
+                }
+            }
+        },
+    };
+
+    // The to-device since token is passed from the previous request.
+    // The extensions haven't changed, so they're not updated (sticky parameters
+    // ftw).
```

Comment:
`assert request` runs a _partial_ assert. Missing data are considered OK.

Edit: it's technically an `Inclusive` comparison, see https://docs.rs/assert-json-diff/latest/assert_json_diff/enum.CompareMode.html#variant.Inclusive.